### ISSUE TITLE
VScode quick fix should not suggest removing `@Autowired` annotation from JUnit tests

### ISSUE THREAD

**Issue by tigerinus** [spring-projects/sts4#786]
**Describe the bug**

After a recent update of Spring Boot extension for VSCode, it starts to suggest removing `@Autowired` annotation from JUnit tests

![image](https://user-images.githubusercontent.com/7172560/173874392-f9a51bde-d979-4148-94c8-633ecfaf6609.png)

However, after the annotation is removed, test is unable to get launched:

```
org.junit.jupiter.api.extension.ParameterResolutionException: No ParameterResolver registered for parameter [org.springframework.test.web.servlet.MockMvc mvc] in constructor [public com.newpathfly.flightsupplier.via.service.SearchServiceIT(org.springframework.test.web.servlet.MockMvc,java.lang.String)].
 at java.util.Optional.orElseGet([Optional.java:267](vscode-file://vscode-app/c:/Users/xiaoh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html))
 at java.util.ArrayList.forEach([ArrayList.java:1259](vscode-file://vscode-app/c:/Users/xiaoh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html))
 at java.util.ArrayList.forEach([ArrayList.java:1259](vscode-file://vscode-app/c:/Users/xiaoh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html))
```

**To Reproduce**

Build a simple Spring Boot test in an existing Spring Boot project:

```java
@SpringBootTest
@AutoConfigureMockMvc
class WhateverIT {
    private final MockMvc _mvc;

    public WhateverIT (MockMvc mvc) {
        _mvc = mvc;
    }

    @Test
    void dummyTest() {
    }
```
> Note that `@Autowired` does not exist.

Then try to launch the test from VSCode

**Comment by BoykoAlex** [comment]
Yes, agreed. Will get this fixed

**Comment by tigerinus** [comment]
> Don't think i have better ideas than skipping the case of a class with `@SpringBootTest` annotation. Any objections?

What's the root cause of the issue?

Would you be able to skip the case of a class with any annotation that `@BootstrapWith(SpringBootTestContextBootstrapper.class)` in case people have their own annotation versus `@SprintBootTest`?

**Comment by BoykoAlex** [comment]
We've added removing @Autowired from the only constructor in class but conditions for this marker and quick fix are buggy.

Yes we can skip class annotated with `@BootstrapWith(SpringBootTestContextBootstrapper.class)` as well without any problems. Thanks for bringing this up

**Comment by BoykoAlex** [comment]
The fix is 382d0bbac672b1180bf6189938491d2786e74512.
We'll be creating another release in a few days from now to include this fix. Once the RC is tested I'll mark it as fixed.

**Comment by BoykoAlex** [comment]
@tigerinus here is the RC that I've tried: https://s3.us-west-1.amazonaws.com/s3-test.spring.io/sts4/vscode-extensions/snapshots/vscode-spring-boot-1.36.0-RC.1.vsix
Seemed to work fine there.
